From 83fd6cbc3cc35d42f764e6cc22878cd4f921f514 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Thu, 28 Aug 2014 22:32:00 -0400
Subject: [drm] i915: silence some scary new warnings

Message-id: <1409265120-10037-4-git-send-email-rclark@redhat.com>
Patchwork-id: 89872
O-Subject: [RHEL7 drm 6/6] drm/i915: stfu
Bugzilla: 1128915
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

RHBZ: 1128915
Upstream: n/a

Recent upstream kernels have added additional checks in pipe state check
code.  Which would be great if i915 actually got these right in all
cases, but that isn't the case yet.  The net result are scary looking
WARN_ON() splats.  These aren't new bugs, and in most cases the user
would not otherwise notice anything.  So to avoid a flood of abrt's,
remove some of these new checks.

Signed-off-by: Rob Clark <rclark@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index f0be855..f729307 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -9966,11 +9966,6 @@ intel_pipe_config_compare(struct drm_device *dev,
 
  PIPE_CONF_CHECK_I(pixel_multiplier);
  PIPE_CONF_CHECK_I(has_hdmi_sink);
- if ((INTEL_INFO(dev)->gen < 8 && !IS_HASWELL(dev)) ||
-     IS_VALLEYVIEW(dev))
-  PIPE_CONF_CHECK_I(limited_color_range);
-
- PIPE_CONF_CHECK_I(has_audio);
 
  PIPE_CONF_CHECK_FLAGS(adjusted_mode.flags,
          DRM_MODE_FLAG_INTERLACE);
-- 
1.7.1